aws_quota/check/ecr.py

```diff
@@ -0,0 +1,43 @@
+from typing import List
+from .quota_check import InstanceQuotaCheck, QuotaScope
+from aws_quota.utils import get_paginated_results
+
+import boto3
+import cachetools
+
+
+@cachetools.cached(cache=cachetools.TTLCache(maxsize=1, ttl=60))
+def get_all_repositories(session: boto3.Session) -> List[str]:
+    services = ['ecr']
+    # ECR is available in all regions, but ecr-public is only available in us-east-1
+    if session.region_name == 'us-east-1':
+        services.append('ecr-public')
+
+    return [
+        repository['repositoryArn'] 
+        for service in services
+        for repository in get_paginated_results(session, service, 'describe_repositories', 'repositories')
+    ]
+
+@cachetools.cached(cache=cachetools.TTLCache(10000, 60))    # 10k = default number of max registries per account
+def get_repository_images(session: boto3.Session, repository_arn: str) -> List[str]:
+    arn_parts = repository_arn.split(':')
+    service = arn_parts[2]
+    repository_name = arn_parts[5].removeprefix("repository/")
+    return get_paginated_results(session, service, "describe_images", "imageDetails", {'repositoryName': repository_name})
```

Comment:
@fheinecke did you try by chance [list-images](https://docs.aws.amazon.com/cli/latest/reference/ecr/list-images.html) call is any quicker?

Reply:
Just tested this. The decrease is negligible and within the margin of error for the tests I ran - around 100us per image. In addition the public ECR endpoint doesn't support list_images, so I'd need to add separate logic for both public and private ECR.